Sources/NIO/ChannelPipeline.swift

```diff
@@ -496,14 +496,12 @@ public final class ChannelPipeline: ChannelInvoker {
     func removeHandlers() {
         assert(eventLoop.inEventLoop)
 
-        if let head = self.head {
-            while let ctx = head.next {
-                remove0(ctx: ctx, promise: nil)
-            }
-            remove0(ctx: self.head!, promise: nil)
+        while let ctx = head.next {
+            remove0(ctx: ctx, promise: nil)
         }
-        self.head = nil
-        self.tail = nil
+
+        self.head.next = tail
```

Comment:
I'm pretty sure this will cause head and tail to never be deallocated as there's a cycle now that I don't think we'll ever break..,